File: wallet/src/main/java/chatox/wallet/event/UserInteractionCreated.java
package chatox.wallet.event;

import chatox.wallet.model.UserInteractionType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.ZonedDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserInteractionCreated {
    private String id;
    private UserInteractionType type;
    private BigDecimal cost;
    private String userId;
    private String targetUserId;
    private ZonedDateTime createdAt;
}

File: user/src/main/kotlin/chatox/user/messaging/rabbitmq/event/UserInteractionCreated.kt
package chatox.user.messaging.rabbitmq.event

import chatox.user.domain.UserInteractionType
import java.math.BigDecimal
import java.time.ZonedDateTime

data class UserInteractionCreated(
        val id: String,
        val type: UserInteractionType,
        val cost: BigDecimal,
        val userId: String,
        val targetUserId: String,
        val createdAt: ZonedDateTime
)